scripts/jenkins/cloud/ansible/crowbar-upgrade.yml

```diff
@@ -0,0 +1,84 @@
+#
+# (c) Copyright 2019 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+---
+
+- name: Upgrade cloud - crowbar
+  hosts: "{{ cloud_env }}"
+  remote_user: root
+  gather_facts: True
+  vars:
+    extra_upgrade_vars:
+      upgrade_progress_file: "/var/lib/crowbar/upgrade/{{ cloud_release_number }}-to-{{ (cloud_release_number | int) + 1 }}-progress.yml"
+      UPDATEREPOS_UPGRADE: "{{ (upgrade_repos is defined and upgrade_repos | length > 0) | ternary(upgrade_repos.split(',') | join('+'), '') }}"
+      cloudsource: "{{ upgrade_cloudsource }}"
+      want_nodesupgrade: 1
+  tasks:
+    - block:
+        - include_role:
+            name: crowbar_setup
+          vars:
+            qa_crowbarsetup_cmd: "{{ upgrade_step.command }}"
+            override_vars: "{{ extra_upgrade_vars|combine(upgrade_step.vars|default({})) }}"
+          loop:
+            - command: onadmin_prepare_cloudupgrade_nodes_repos
```

Comment:
Hm, aren't you missing change of `cloudsource` here?
See the execution in
https://github.com/SUSE-Cloud/automation/blob/master/scripts/mkcloud#L1105

Reply:
Further testing revealed that you were right, even though the playbook is using the default `cloudsource` value here. The way I was inserting the env variable overrides into qa_crowbarsetup.sh was incorrect. I fixed it: now I'm simply updating `mkcloud.config` with the env var overrides.